packages/node_modules/@node-red/editor-client/src/js/ui/tab-info-outliner.js

```diff
@@ -386,11 +391,20 @@ RED.sidebar.info.outliner = (function() {
     function onObjectRemove(n) {
         var existingObject = objects[n.id];
         existingObject.treeList.remove();
-        delete objects[n.d]
+        delete objects[n.id]
         var parent = existingObject.parent;
         if (parent.children.length === 0) {
             parent.treeList.addChild(getEmptyItem(parent.id));
         }
+        if (existingObject.children && (existingObject.children.length > 0)) {
```

Comment:
Is this related to the 'deleting a group' issue? It looks like this code moves the child items up to the parent... but what is the scenario that needs that? Deleting a group deletes the children, so I don't see how this code helps that case.

Reply:
Deletion of a group triggers the deletion event in turn to nodes.
Thus, if the connection relation is not corrected,  a problem may occur in the process of deleting the children.

Reply:
I still can't see what scenario it is trying to handle. When a group is deleted, all of the nodes are deleted first, then the group. It is never the case that `existingObject.children.length >0` in that scenario.

I have tested deleting subflows, nodes, groups - at no time is this code run.

It *does* get run when switching projects and throws an error because it tries to reparent nodes inside a subflow to the top-level Subflow section of the tree, which is just wrong (and breaks because there is no `id` for that parent object).

I realise this was over a week ago, but did you hit a scenario that *needed* this code?

Reply:
You can reproduce the error by following steps:
1. Importing the sample flow at the beginning of this thread
2. fully expand outliner tree as follows:
![スクリーンショット 2020-06-03 11 16 22](https://user-images.githubusercontent.com/30289092/83588425-50556680-a58c-11ea-88cb-1dce706f2c59.png)
3. delete the outermost group by selecting it then pressing delete key
4. following error occurs in browser:
```
RED.events.emit error: [groups:remove] Error: cannot call methods on editableList prior to initialization; attempted to call method 'removeItem'
```

Reply:
Thanks - its the nested groups that I was missing.

Whilst investigating further, I found a few other cases that were not being handled - such as deleting a tab containing nested groups.

I've pushed https://github.com/node-red/node-red/commit/efecfa328bcee4a0af3f48a0422e202e7de9747a which fixes all of those cases. It often came down to the order things were being deleted - getting the order of the events right is critical to keeping the logic correct.